src/css3/new_tokenizer.rs

```diff
@@ -77,6 +103,62 @@ pub enum Token {
     EOF,
 }
 
+impl Token {
+    pub fn kind(&self) -> TokenKind {
+        match self {
+            Token::AtKeyword(_) => TokenKind::AtKeyword,
```

Comment:
I suspect we can turn the `Token` struct into an enum.  I do not know what unwanted side effects this would have.

```rust
pub enum Token {
  At(String),
  Hash(String),
  Dot(String),
  // ...
}
```

If you wanted a more complex payload (e.g., to track line and column numbers), you could replace `String` with `Inner`:

```rust
pub struct Inner {
  pub value: String,
  pub line: usize,
  pub col: usize,
}

pub enum Token {
  At(Inner),
  Hash(Inner),
  Dot(Inner),
  // ...
}
```

If you ever needed a different kind of struct for one of the tokens, you could replace `Inner` with something more specific in that case.

I do not think we'd need a separate `kind` enum unless we ran into an unusual corner case with serialization or something, but I could be wrong.

You might discover a reason that this approach would not make sense.

Reply:
@neuodev it occurred to me that you might be using the `kind` as a state machine, similar to [the one defined here](https://github.com/gosub-browser/gosub-engine/blob/69311995858fb02eceee7ff2883fb421dc1f8c16/src/html5/tokenizer/state.rs#L3).  If that is its purpose, then I don't think my comment above necessarily makes sense.  Please ignore that comment if it's not applicable.